Bindings/Java/Matlab/Utilities/osimTableFromStruct.m

```diff
@@ -20,108 +26,91 @@
 % permissions and limitations under the License.                          %
 % ----------------------------------------------------------------------- %
 
-%% Convert Matlab Struct to OpenSim time Series Table
-%  Input is a Maltab stucture where data.label = nX1 or nX3 array
-%  eg structdata.LASI = [2 3 4; 4 5 6, ...
-%  One of the structures values MUST be a time vector and called 'time'
-%  Output is an OpenSim TimesSeriesTable
-
-% Author: James Dunne, Tom Uchida, Shrinidhi K. Lakshmikanth, Chris Dembia, 
-% Ajay Seth, Ayman Habib, Jen Hicks.
+% Written by: James Dunne, Tom Uchida, Chris Dembia, Ajay Seth,
+%                   Ayman Habib, Jen Hicks,Shrinidhi K. Lakshmikanth.
 
-function timeseriesosimtable = osimTableFromStruct(structdata)
-% import Java Libraries
+function timeseriesosimtable = osimTableFromStruct(s)
+%% import Java Libraries
 import org.opensim.modeling.*
 
-% Each column of data has a label. We wiget all the data labels
-labelnames = fieldnames(structdata);
-% get the index for the time array
-timeIndex = find(cellfun(@(s) ~isempty(strfind('time', s)),lower(labelnames)));
+%% Get all the data labels in the Structure
+labels = fieldnames(s);
 
-%% Check the dimensions of the structure confirm to requirements
-if isempty(timeIndex)
-    error('Time field required ? none found')
+%% Check for, save, then remove the time array from the structure
+% Get the index for time array
+tIndex = find(cellfun(@(s) contains('time', s),lower(labels)));
+% Check to see if the time exists
+if isempty(tIndex)
+    error('"time" field required, none found')
 end
-
+% Get the time vector
+time = s.time();
+% Remove time from the struct
+s = rmfield(s,'time');
```

Comment:
I'm not sure I understand how the previous code exhibited a bug.

Reply:
It worked fine as long as the 'time' field was the last field in the struct. If you added more fields, I did a poor job of dealing with indexes. Also, the code was so poorly written that it took me ages to figure out exactly where the index issue was occurring, so that is a sign I should improve it. Taking the time out of the Structure here just makes everything easier.